### PR TITLE
feat(server):add search api

### PR DESCRIPTION
**Pull request by piyushagwl007**
## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-9-Lumen/wiki/Commit-message-convention)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (only for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Other... Please describe:
Issue Number: 
#85 

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: server/src/services/blogs/service.js [hunk 1]
```diff
@@ -26,9 +26,26 @@ const deleteBlogById = async (blogID) => {
   return blog.toObject();
 };
 
+const getBlogsForQuery = async (page, categories, searchString) => {
```
Comment (nimish-gupta) [inline server/src/services/blogs/service.js]: Tests for this service

**Review by nimish-gupta — CHANGES_REQUESTED** [review]

**Review by karansapolia — APPROVED** [review]
LGTM, having the service test will be helpful

File: server/src/services/home/api.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const health = (req, res) => {
```
Comment (nimish-gupta) [inline server/src/services/home/api.js:1]: Why you have health route for every type of router?

**Review by nimish-gupta — APPROVED** [review]

**Review by nihalanisumit — APPROVED** [review]
LGTM

File: server/src/services/home/api.js (hunk 2 above)
Comment (piyushagwl007) [reply server/src/services/home/api.js:1]: To check if the service is available/reachable or not :smile:

**Pull request merged**